### PR TITLE
Redis Desktop Manager 0.9.9 (new formula)

### PR DESCRIPTION
**Pull request by jameshilliard**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Reopening since [last pull request](https://github.com/Homebrew/homebrew-core/pull/38569) was stale.

### CODE REVIEW

**Comment by zbeekman** [comment]
I'd love to see this in Homebrew, but I do agree that this is more appropriate as a cask. Any chance we could convince upstream to create binaries for this? I don't think it's appropriate for Homebrew-core. 

An alternate approach would be to create your own tap. Or create a repo that builds the app (possibly based on your work in this PR) and publishes it. Then create a cask from the published app.

**Comment by jameshilliard** [comment]
> Any chance we could convince upstream to create binaries for this? I don't think it's appropriate for Homebrew-core.

Nope, upstream won't provide binaries.

> An alternate approach would be to create your own tap. Or create a repo that builds the app (possibly based on your work in this PR) and publishes it. Then create a cask from the published app.

I don't think that would really work since I would have to then build/distribute the binaries.

File: Formula/rdm.rb
Comment (zbeekman) [inline Formula/rdm.rb]: Definitely don't want to be putting an `.app` in the Cellar.

**Comment by zbeekman** [comment]
From https://redisdesktop.com/download

>  Subscription required. Binaries are available only for users with subscription.


@jameshilliard:
> > An alternate approach would be to create your own tap. Or create a repo that builds the app (possibly based on your work in this PR) and publishes it. Then create a cask from the published app.
> 
> I don't think that would really work since I would have to then build/distribute the binaries.

I'm not sure there's much other recourse here, but let's have other @Homebrew/maintainers weigh in here. I will note that upstream seems to be intentionally not shipping binaries because that seems to be part of their business model, but, they have released the software under GPL V3, so we are free to build and distribute binaries.

Otherwise, you can take a look at https://github.com/brewsci/homebrew-num to see how you can setup a tap that will automatically build and deploy bottles. (Also check https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.) You can use Travis-CI or Azure pipelines to do the app build and deployment.

File: Formula/rdm.rb
Comment (zbeekman) [inline Formula/rdm.rb]: This test is not acceptable. A test is required that actually exercises some functionality.

**Review by zbeekman — CHANGES_REQUESTED** [review]
If we include this, it will need a better test, and we'll have to consider if/where to put the app bundle.

File: Formula/rdm.rb
Comment (jameshilliard) [reply Formula/rdm.rb]: How is using the `.app` here different from say [this](https://github.com/Homebrew/homebrew-core/blob/f5b2a69e7659bfcbdeb3fb0bc5a7bd526b143158/Formula/tarsnap-gui.rb#L22)?

File: Formula/rdm.rb
Comment (jameshilliard) [reply Formula/rdm.rb]: I see a bunch of other formulas doing this. For example [here](https://github.com/Homebrew/homebrew-core/blob/f5b2a69e7659bfcbdeb3fb0bc5a7bd526b143158/Formula/tarsnap-gui.rb#L26). Isn't this enough to verify that there is an executable binary?

**Comment by jameshilliard** [comment]
I tried to follow the same conversions used by other homebrew-core `.app` style formulas.

File: Formula/rdm.rb
Comment (zbeekman) [reply Formula/rdm.rb]: Just because other formulas are doing it doesn't mean we want to accept new formulas using bad older conventions. If there is a *very* compelling case to add a new formula and a *very* compelling case as to why there is no possibility of more thorough testing, then we will *consider* adding a new formula without more advanced testing. However, unless you can demonstrate that there is no other feasible way to test for correct functionality, this test is a non-starter for us, and the formula won't be accepted unless the test is improved.

Comment (jameshilliard) [reply Formula/rdm.rb]: These are the command line options:
```
$ rdm --help
Usage: rdm [options]

Options:
  -h, --help                              Displays this help.
  -v, --version                           Displays version information.
  --settings-dir <settingsDir>            (Optional) Directory where RDM
                                          looks/saves .rdm directory with
                                          connections.json file
  --formatters-dir <formattersDir>        (Optional) Directory where RDM looks
                                          for native value formatters
  --rendering-backend <renderingBackend>  (Optional) QML rendering backend
                                          [software|opengl|d3d12|'']
```

File: Formula/rdm.rb
Comment (zbeekman) [reply Formula/rdm.rb]: OK, upon further inspection I see that we do have a number of apps being installed (32) out of 4771 total formulae (0.67%), and if we install an App, it does go in that location.

I'm still unsure as to whether we are willing to accept new formulae that build apps at this time, and will wait for feedback from other maintainers.

File: Formula/rdm.rb
Comment (zbeekman) [reply Formula/rdm.rb]: You may be able to spin up an instance of the app with the `--settings-dir`, option set, then send a signal to shutdown and then check for the existence, and, possibly the contents, of the `connections.json` file. You can also look for some correct logging to stdout/stderr, if it's suitably general and not likely to change between versions, or at any log files written elsewhere by the app bundle.

**Comment by zbeekman** [comment]
Can other @Homebrew/maintainers weigh in here? Are we willing to add a new formula that builds an .app bundle? After reviewing this, I am personally 👍 for adding this formula, but I know we typically try not to include app bundles in core. In this case it seems that upstream, while licensing under GPLv3, is trying to profit from shipping binaries. Obviously we are free to build and distribute binaries under GPLv3. This may annoy upstream, but I would like to see OSI licensed software easily and freely ($) installable, which is why I am 👍 on this case.

**Comment by MikeMcQuaid** [comment]
> Are we willing to add a new formula that builds an .app bundle?

No: https://docs.brew.sh/Acceptable-Formulae#stuff-that-builds-an-app. This is why `brew linkapps` was removed. A timely reminder from 10 years ago: https://github.com/Homebrew/legacy-homebrew/commit/29d85578e75170a6c0eaebda4d701b46f1acf446#diff-c47c7c7383225ab55ff591cb59c41e6bR90

`brew cask` installs a bunch of stuff that requires a license to work. Additionally, a third-party tap could be created for installing a formula. This formula is involved, replicating work upstream and unable to have a good test.

**Comment by zbeekman** [comment]
Thanks for weighing in Mike. Apparently I'm blind because I was looking for that in Acceptable-Formulae, but didn't see it... sigh.

**Comment by zbeekman** [comment]
@jameshilliard Thanks for your work and persistence. Sorry, but unfortunately we are unable to accept the redis-desktop formula into homebrew-core.

**Comment by jameshilliard** [comment]
Maybe it makes sense to have a source built cask repo for these sorts of
projects under the Homebrew umbrella?

**Comment by MikeMcQuaid** [comment]
> Maybe it makes sense to have a source built cask repo for these sorts of projects under the Homebrew umbrella?

@jameshillard Good idea. If you create one we'll accept a PR to add a link to https://docs.brew.sh/Interesting-Taps-and-Forks

**Comment by jameshilliard** [comment]
@MikeMcQuaid something like https://github.com/Homebrew/homebrew-cask-build?

**Comment by MikeMcQuaid** [comment]
@jameshilliard It will need to be a tap outside the Homebrew organisation.

**Comment by jameshilliard** [comment]
@MikeMcQuaid So homebrew doesn't plan to support `.app`'s anymore I guess? Does that mean the existing ones in homebrew-core should be removed?

**Comment by MikeMcQuaid** [comment]
> @MikeMcQuaid So homebrew doesn't plan to support formulas for `.app`'s anymore I guess?

Yes.

> Does that mean the existing ones in homebrew-core should be removed?

We have a higher bar for removing existing formulae than adding new ones so: maybe.

**Comment by zbeekman** [comment]
If we end up having a dedicated tap for apps built from source, then I don't see any reason why we wouldn't migrate the existing formulae that build apps to the new tap.

**Comment by MikeMcQuaid** [comment]
> If we end up having a dedicated tap for apps built from source, then I don't see any reason why we wouldn't migrate the existing formulae that build apps to the new tap.

It would need to be a community run tap outside the Homebrew organisation. As a result we may not migrate as it would harm discoverability.

**Comment by jameshilliard** [comment]
> It would need to be a community run tap outside the Homebrew organisation. As a result we may not migrate as it would harm discoverability.

Without the homebrew build/test infrastructure it wouldn't be nearly as useful I don't think since binaries wouldn't easily be able to be provided.

**Comment by MikeMcQuaid** [comment]
Regardless: we are not going to introduce additional formula taps in the Homebrew organisation as we have historically had a bad experience with them.

**Comment by jameshilliard** [comment]
Would this be suitable if I could modify it to not install an `.app`?

**Comment by MikeMcQuaid** [comment]
If it has a commandline, non-GUI mode: yes, maybe. If it's just a GUI without an `.app`: no, sorry.

**Pull request closed without merging**
